stablehlo/transforms/Passes.td

```diff
@@ -258,3 +258,34 @@ def StablehloLegalizeQuantizedOpToQDQPass : Pass<"stablehlo-legalize-quantized-o
     "mlir::stablehlo::StablehloDialect",
   ];
 }
+
+def StablehloLegalizeQDQToQuantizedOpPass : Pass<"stablehlo-legalize-qdq-to-quantized-op", "mlir::func::FuncOp"> {
+  let summary = "Compose StableHLO quantized op using uniform quantize/dequantize ops";
+
+  let description = [{
+    Compose StableHLO quantized op using uniform quantize/dequantize ops.
+    Note: The pass does not delete any preexisting op.
+    For example, the following program
+
+    ```mlir
+    func.func @add(%arg0: tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>) -> tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>> {
+      %0 = stablehlo.uniform_dequantize %arg0 : (tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>) -> tensor<16x16xf32>
+      %1 = stablehlo.abs %0 : tensor<16x16xf32>
+      %2 = stablehlo.uniform_quantize %1 : (tensor<16x16xf32>) -> tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
+      func.return %2 : tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
+    }
+    ```
+
+    Will become:
+
+    ```mlir
+    func.func @add(%arg0: tensor<16x16x!quant.uniform<u8:f32, 3.400000e+01:16>>) -> tensor<16x16x!quant.uniform<u8:f32, 3.400000e+01:16>> {
+      %0 = stablehlo.abs %arg0 : tensor<16x16x!quant.uniform<u8:f32, 3.400000e+01:16>>
```

Comment:
Is there a reason we are not showing the original non-deleted ops?

Reply:
to keep the example short and focus on what pass does. Description notes that pass doesn't delete the preexisting ops. discussed [here ](https://github.com/openxla/stablehlo/pull/2478#discussion_r1707769311)

Reply:
closing this as we are discussing on at another comment thread.